addons/pkg-npm/jaxrs/src/main/java/org/commonjava/indy/pkg/npm/jaxrs/PackageContentAccessHandler.java

```diff
@@ -0,0 +1,425 @@
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.pkg.npm.jaxrs;
+
+import org.apache.commons.io.IOUtils;
+import org.commonjava.indy.IndyWorkflowException;
+import org.commonjava.indy.content.ContentManager;
+import org.commonjava.indy.core.bind.jaxrs.ContentAccessHandler;
+import org.commonjava.indy.core.bind.jaxrs.util.TransferStreamingOutput;
+import org.commonjava.indy.model.core.PackageTypes;
+import org.commonjava.indy.model.core.StoreKey;
+import org.commonjava.indy.model.core.StoreType;
+import org.commonjava.indy.model.util.HttpUtils;
+import org.commonjava.indy.pkg.npm.content.group.PackageMetadataMerger;
+import org.commonjava.indy.util.AcceptInfo;
+import org.commonjava.indy.util.ApplicationContent;
+import org.commonjava.indy.util.ApplicationHeader;
+import org.commonjava.maven.atlas.ident.util.VersionUtils;
+import org.commonjava.maven.galley.event.EventMetadata;
+import org.commonjava.maven.galley.model.Transfer;
+import org.commonjava.maven.galley.model.TransferOperation;
+import org.commonjava.maven.galley.transport.htcli.model.HttpExchangeMetadata;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Date;
+import java.util.function.Consumer;
+
+import static org.commonjava.indy.bind.jaxrs.util.ResponseUtils.formatOkResponseWithEntity;
+import static org.commonjava.indy.bind.jaxrs.util.ResponseUtils.formatResponse;
+import static org.commonjava.indy.bind.jaxrs.util.ResponseUtils.formatResponseFromMetadata;
+import static org.commonjava.indy.bind.jaxrs.util.ResponseUtils.setInfoHeaders;
+
+public class PackageContentAccessHandler
+                extends ContentAccessHandler
+{
+    public static final String CROSS = "-";
+
+    public static final String PATH_SPLITER = "/";
+
+    public static final String TGZ = ".tgz";
+
+    public static final String PACAGE_TGZ = "package" + TGZ;
+
+    @Override
+    public Response doHead( final String packageType, final String type, final String name, final String packageName,
+                            final Boolean cacheOnly, final String baseUri, final HttpServletRequest request,
+                            EventMetadata eventMetadata, final Consumer<Response.ResponseBuilder> builderModifier )
+    {
+        return doHead( packageType, type, name, packageName, null, null, cacheOnly, baseUri, request, eventMetadata,
```

Comment:
i think it's okay to hard-code the package type into this, and avoid the parameter. I would expect something similar to this for other package types, so I don't think generalizing this one will help too much. We can always generalize it later, since it's not actually implementing the REST API directly.